### PR TITLE
New UI

### PR DESCRIPTION
**Pull request by UnbearableBear**
I think it's better to pull everything and review locally than to check all the files.
Still, the commit are split so you can read them step by step

For now, only Button, theme and layout components are migrated. The remaining component will be migrated in a following PR

FEATURES:

- Accessibility check in the Storybook
- Themed colors with ThemeProvider
- Not a single hack !
- Soon to come netlify version of the static storybook

Most interesting commit: https://github.com/SocialGouv/code-du-travail-numerique/pull/1433/commits/fde922677e6395d49761446079b9674e74498550

Most interesting file: https://github.com/SocialGouv/code-du-travail-numerique/pull/1433/commits/fde922677e6395d49761446079b9674e74498550#diff-3edb2f689c6750ee71822776faa31f7f

Screenshot 2019-10-09 at 16 54 06
Screenshot 2019-10-09 at 16 54 11
Screenshot 2019-10-09 at 16 54 57

### CODE REVIEW

File: packages/code-du-travail-frontend/static/assets/icons/external-link.svg [hunk 1]
```diff
@@ -1 +1,10 @@
-<svg xmlns="http://www.w3.org/2000/svg" width="13.93cm" height="13.4cm" viewBox="0 0 394.91 379.91"><defs><style>.cls-1{fill:#4c5467;}</style></defs><title>Plan de travail 1</title><path class="cls-1" d="M26.06,157.2A52.18,52.18,0,0,1,78.18,105h95.56V79H78.18A78.27,78.27,0,0,0,0,157.16V301.73a78.27,78.27,0,0,0,78.18,78.18H243.72a78.27,78.27,0,0,0,78.18-78.18V222H295.83v79.75a52.18,52.18,0,0,1-52.13,52.13H78.18a52.18,52.18,0,0,1-52.12-52.13"/><path class="cls-1" d="M196.07,12.32A13,13,0,0,0,208.32,26h0L342.83,33.6,164.26,212.22a13,13,0,0,0,18.43,18.43h0L361.27,52.07l7.63,134.49a13,13,0,1,0,26-1.47h0L386.63,39A32.56,32.56,0,0,0,355.9,8.27L209.81,0a13,13,0,0,0-13.74,12.21Z"/></svg>
\ No newline at end of file
+<svg viewBox="0 0 394.91 379.91">
+  <path 
+    fill="currentColor"
```
Comment (UnbearableBear) [inline packages/code-du-travail-frontend/static/assets/icons/external-link.svg:3]: failed attempt to make the icon get the currentColor

Comment (revolunet) [reply packages/code-du-travail-frontend/static/assets/icons/external-link.svg:3]: 😂

**Comment by revolunet** [comment]
smoothest migration ever

**Review by nkrmr — APPROVED** [review]

File: packages/react-ui/.out/iframe.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
+<!doctype html><html lang="en"><head><meta charset="utf-8"><title>Storybook</title><meta name="viewport" content="width=device-width,initial-scale=1"><base target="_parent"><style>:not(.sb-show-main) > .sb-main,
```
Comment (lionelB) [inline packages/react-ui/.out/iframe.html:1]: Does `.out` need to be versionned ?

File: packages/react-ui/.storybook/config.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      components: [ThemeProvider],
+      params: [
+        { name: "Default Theme", props: { theme: colors }, default: true },
+        { name: "Inverted Theme", props: { theme: invertedColors } }
+      ]
+    }
+  ])
+);
+addDecorator(withA11y);
+// https://github.com/dequelabs/axe-core/blob/develop/doc/API.md#api-name-axeconfigure
```
Comment (lionelB) [inline packages/react-ui/.storybook/config.js]: should we keep that comment ?

Comment (UnbearableBear) [reply packages/react-ui/.storybook/config.js]: I'll remove it !

File: packages/react-ui/.out/iframe.html (hunk 2 above)
Comment (UnbearableBear) [reply packages/react-ui/.out/iframe.html:1]: Yes, netlify will watch this folder and update the static website everytime new files are pushed

File: packages/code-du-travail-ui-old/src/Toast/taist.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from "react";
 import { fireEvent, render } from "@testing-library/react";
 import Toast from "../Toast";
 
+// unusable test for now because of the button import
```
Comment (lionelB) [inline packages/code-du-travail-ui-old/src/Toast/taist.js]: it could be better to use `testPathIgnorePatterns`

Comment (UnbearableBear) [reply packages/code-du-travail-ui-old/src/Toast/taist.js]: Next PR problem is auto solved, they will both be in the same UI. I will uncomment it, it's a promess

Comment (UnbearableBear) [reply packages/code-du-travail-ui-old/src/Toast/taist.js]: done !

File: packages/react-ui/.storybook/config.js (hunk 3 above)
Comment (UnbearableBear) [reply packages/react-ui/.storybook/config.js]: done

File: packages/react-ui/.out/iframe.html (hunk 2 above)
Comment (revolunet) [reply packages/react-ui/.out/iframe.html:1]: Netlify Can also build so we dont need to version the build (see emjpm-ui)

Comment (UnbearableBear) [reply packages/react-ui/.out/iframe.html:1]: Cool, I haven't really looked at it yet. ! I'm waiting to finish the migration

**Pull request merged**